src/routes/safe/components/Apps/components/AddAppForm/AppAgreement.tsx

```diff
@@ -9,27 +9,45 @@ import Field from 'src/components/forms/Field'
 const StyledCheckbox = styled(Checkbox)`
   margin: 0;
 `
+const StyledAgreementContainer = styled.div`
+  display: flex;
+  flex: 1;
+  .MuiFormControlLabel-root {
+    margin-right: 0;
+  }
+`
 
 const AppAgreement = (): React.ReactElement => {
   const { visited } = useFormState({ subscription: { visited: true } })
-
+  const [checked, setCheck] = useState(false)
   // trick to prevent having the field validated by default. Not sure why this happens in this form
   const validate = !visited?.agreementAccepted ? undefined : required
 
+  const handleOnChange = ({ onBlur, onFocus, onChange }, event, value: boolean) => {
+    onFocus(event)
+    onChange(event)
+    onBlur(event)
+    setCheck(value)
+  }
+
   return (
-    <Field
-      component={StyledCheckbox}
-      label={
-        <Text size="xl">
-          This app is not a Gnosis product and I agree to use this app
-          <br />
-          at my own risk.
-        </Text>
-      }
-      name="agreementAccepted"
-      type="checkbox"
-      validate={validate}
-    />
+    <Field name="agreementAccepted" type="checkbox" validate={validate}>
```

Comment:
This solution is too ad-hoc. 
I would create a new component `FinalFormCheckbox` and make the arrangement needed there, as we probably will need to face this problem in another part of the app.